src/courseware/data/selectors.js

```
@@ -1,20 +1,22 @@
 /* eslint-disable import/prefer-default-export */
 export function sequenceIdsSelector(state) {
-  if (state.courseware.courseStatus !== 'loaded') {
+  try {
```

These catch-all try/catch blocks make me nervous... both because they assume an error will be thrown, and because they swallow all errors that may occur.

---

Fair enough about the swallowing of errors. I did this rather than adding more checks. I had half a mind to add lodash.get so we don't need to check that everything exists first. Should I take the check that things are not null route?

---

The `state.courseware.courseStatus !== 'loaded'` check was a way of avoiding all those "not null" checks, in that if the course isn't loaded, we just avoid the whole thing because we know the data won't be there.  Is the problem that we can now say the course is "loaded" even though the course blocks API has failed?  I think that might be a problem if so... it makes "loaded" kinda meaningless, rather than definitive.  If both of those APIs aren't successful, we should be dead in the water and just fail fast, I think.